### PR TITLE
feat: Add Psalm integration

### LINKED ISSUE #5: Psalm integration
### Feature Request

|    Q        |   A
|------------ | ------
| QA    | yes

#### Summary

As decided during the [Technical-Steering-Committee Meeting on August 3rd, 2020](https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-08-03-TSC-Minutes.md#vote-on-static-analysis-tooling), Laminas wants to implement [vimeo/psalm](https://github.com/vimeo/psalm) in all packages.

Implementing psalm is quite easy.

#### Required
- [ ] Create a `.psalm.xml.dist` in the project root
- [ ] Copy and paste the contents from this [psalm.xml.dist](https://github.com/mezzio/repo-template/blob/6124090610ef1c6d6b0c80a063cafacc3ec68f38/psalm.xml.dist)
- [ ] Run `$ composer require vimeo/psalm`
- [ ] Run `$ vendor/bin/psalm --set-baseline=psalm-baseline.xml`
- [ ] Add a composer script `static-analysis` with the command `psalm --shepherd --stats`
- [ ] Add a new line to `script:` in `.travis.yml`: `- if [[ $TEST_COVERAGE == 'true' ]]; then composer static-analysis ; fi`
- [ ] Remove phpstan from the project (`phpstan.neon.dist`, `.travis.yml` entry, `composer.json` `require-dev` and `scripts`)

##### Optional
- [ ] Fix as many psalm errors as possible.

### PR DESCRIPTION
**Pull request by alexraputa**
Closes #5

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### CODE REVIEW

File: src/CreateHandler/CreateTemplate.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function generateTemplate(
         );
     }
 
-    private function resolveRendererType(?string $templateSuffix): string
+    private function resolveRendererType(?string $templateSuffix): ?string
```
Comment (Ocramius) [inline src/CreateHandler/CreateTemplate.php:104]: Is there a way to maintain a stricter return type here? 🤔

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/CreateHandler/CreateTemplate.php (hunk 1 above)
Comment (alexraputa) [reply src/CreateHandler/CreateTemplate.php:104]: https://github.com/mezzio/mezzio-tooling/blob/04adfc1564bd6bdc29d5041e25bcce309b7ef807/src/CreateHandler/CreateTemplate.php#L110
https://github.com/mezzio/mezzio-tooling/blob/04adfc1564bd6bdc29d5041e25bcce309b7ef807/src/TemplateResolutionTrait.php#L87-L94
@Ocramius Can we throw exception if `$type` is null?

Comment (alexraputa) [reply src/CreateHandler/CreateTemplate.php:104]: Or should I revert these changes?

Comment (Ocramius) [reply src/CreateHandler/CreateTemplate.php:104]: Looking back at this, this is `private` API, so we're fine as-is :+1:

**Review by Ocramius — APPROVED** [review]
:ship:

**Comment by Ocramius** [comment]
Thanks @alexraputa!

**Pull request merged**